src/entry.rs

```diff
@@ -17,25 +19,30 @@ macro_rules! entry {
             panic!("Out of memory")
         }
 
+        #[cfg(not(target_arch = "riscv64"))]
         #[no_mangle]
         pub extern "C" fn _start() -> ! {
-            #[cfg(target_arch = "riscv64")]
-            unsafe {
-                asm!(
-                    "lw a0, 0(sp)",
-                    "addi a1, sp, 8",
-                    "li a2, 0",
-                    "call {}",
-                    "li a7, 93",
-                    "ecall",
-                    sym $main,
-                    options(noreturn)
-                );
-            }
-            #[cfg(not(target_arch = "riscv64"))]
-            panic!("This macro is only valid for riscv64 target");
+            panic!("ckb_std::entry is only valid for riscv64 target")
         }
 
+        // Use global_asm so the compiler won't insert function prologue in _start.
+        #[cfg(target_arch = "riscv64")]
+        core::arch::global_asm!(
+            ".global _start",
+            "_start:",
+            // Argc.
+            "lw a0, 0(sp)",
+            // Argv.
+            "addi a1, sp, 8",
+            // Envp.
+            "li a2, 0",
```

Comment:
What is the purpose of setting a2 = 0?

Reply:
Copied from original code, which seems to be copied from assembly generated by compiling C code.

Reply:
It's weird, but if the code works, leave it as it is.

Reply:
Yeah, it's envp:

https://github.com/riscvarchive/riscv-newlib/blob/f289cef6be67da67b2d97a47d6576fa7e6b4c858/libgloss/riscv/crt0.S#L51-L53